### PR TITLE
docs: Add support policy and update release doc

### LINKED ISSUE #125: Document alignment with supported Cluster API and Kubernetes versions
Let's add a section to the README that describes which versions of Cluster API work with which versions of this repository.

Let's also add a section that describes which versions of Kubernetes this provider is able to provision.

**Example:**
This provider's versions are compatible with the following versions of Cluster API:

||Cluster API 0.1|Cluster API 0.2|
|-|-|-|
|XYZ Provider 0.1|✓||
|XYZ Provider 0.2||✓|

This provider's versions are able to install and manage the following versions of Kubernetes:

||Kubernetes 1.11|Kubernetes 1.12|Kubernetes 1.13|Kubernetes 1.14|
|-|-|-|-|-|
|XYZ Provider 0.1|✓|✓|✓|
|XYZ Provider 0.2||✓|✓|✓|

/kind documentation
/milestone v1alpha1
/priority important-soon

### PR DESCRIPTION
**Pull request by justaugustus**
Signed-off-by: Stephen Augustus <saugustus@vmware.com>

**What this PR does / why we need it**:
Define a support policy for the project.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #125 

**Special notes for your reviewer**:
Initially held for reviews from other cap* maintainers.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by justaugustus** [comment]
I'd like to get a review for other cap* maintainers before merging this.

/hold
/assign @tariq1890 @soggiest @ncdc @detiber @randomvariable @chuckha 
/priority important-soon
/milestone v1alpha1
/kind documentation

File: README.md [hunk 1]
```diff
@@ -24,7 +24,31 @@ cluster on Azure.
...
+This provider's versions are able to install and manage the following versions of Kubernetes:
+
+||Kubernetes 1.13|Kubernetes 1.14 (unreleased)|
+|-|-|-|
+|Azure Provider 0.1 (unreleased)|✓|? (Currently untested)|
+|Azure Provider 0.2 (unreleased)|x|✓|
+
+In general, Cluster API for Azure will remain in lock-step with Cluster API versioning in terms of support e.g., Azure Provider `v0.x` will maintain compatibility with Cluster API `v0.x`.
+
+Each version of Cluster API for Azure will attempt to support two Kubernetes versions e.g., Cluster API for Azure `v0.1` may support Kubernetes 1.13 and Kubernetes 1.14.
```
Comment (chuckha) [inline README.md]: in practice this will probably be _at least_ 3 versions :D, but I think it's fine to be safe while all of this is still playing out

Comment (justaugustus) [reply README.md]: Yep. I added a note in our release documentation to assess this and potentially update it as required each time we release.

**Comment by justaugustus** [comment]
Thanks @ncdc and @chuckha!
/hold cancel

**Pull request merged**